### PR TITLE
feat(app:config:dump): commerce fixes #474

### LINKED ISSUE #474: Add Commerce Command: app:config:dump
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Adding the app:config:dump command to commerce, bin/magento directory. 
This command should be able to take an array of `config types` as an argument

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by rwalter215**
Added app:config:dump command, updated unit tests, and added babel-eslint so we can use static properties in classes. 
## Description

* Added app:config:dump command
* updated unit tests for dump and import
* added babel-eslint to dev-dependencies so we can use static properties in classes
* updated .eslintrc to use babel-eslint as parser

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+governing permissions and limitations under the License.
+*/
+
+const BaseCommerceCliCommand = require('../../../../../../base-commerce-cli-command')
+const { getProgramId } = require('../../../../../../cloudmanager-helpers')
+const commonFlags = require('../../../../../../common-flags')
+const commonArgs = require('../../../../../../common-args')
+
+class AppConfigDumpCommand extends BaseCommerceCliCommand {
+  static strict = false
```
Comment (rwalter215) [inline src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js:19]: This is the reason I added `babel-eslint`. This is the solution that oclif documentation provides for variable length args, but it errored out on `npm run lint-fix`. After researching it appears that eslint does not support static properties on classes out of the box. 

oclif documentation: https://oclif.io/docs/args
^ excerpt I am referring to: "For variable length arguments, disable argument validation with static strict = false on the command."

Comment (justinedelson) [reply src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js:19]: I think you can set static on the class and it accomplishes the same thing. Used here: https://github.com/adobe/aio-cli-plugin-cloudmanager/blob/main/src/commands/cloudmanager/ip-allowlist/bind.js#L42 (amongst other places).

But... if babel-eslint is really necessary, shouldn't `@babel/eslint-parser` be used? Per https://github.com/babel/babel-eslint that's the new location.

File: src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  async run () {
+    const { args, flags, argv } = this.parse(AppConfigDumpCommand)
+
+    const programId = getProgramId(flags)
+    const configTypes = argv.slice(1)
```
Comment (justinedelson) [inline src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js:25]: also... do we need a check here that the argv array length is > 1

Comment (rwalter215) [reply src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js:25]: I don't believe so, if args are length 1, then configTypes with be an empty array and on line 31, options property will be `["-n"]` , which is the desired behavior.

File: src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js (hunk 1 above)
Comment (rwalter215) [reply src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js:19]: Thanks, you are correct. `AppConfigDumpCommand.strict = false` did the trick

Comment (rwalter215) [reply src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js]: I pushed up the requested change, this is resolved

File: src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js (hunk 2 above)
Comment (justinedelson) [reply src/commands/cloudmanager/commerce/bin-magento/app/config/dump.js:23]: Ah... and argv can never be length 0. Got it. Sorry for the noise.

**Pull request merged**